app/javascript/components/admin/UserModal.js

```diff
@@ -71,69 +78,108 @@ class UserModal extends React.Component {
                 }
                 options={this.props.roles.map(role => {
                   return { label: role, value: role };
                 })}
-                onChange={this.handleRoleChange}
+                onChange={data => this.setState({ roleTitle: data.value })}
                 placeholder=""
                 styles={cursorPointerStyle}
                 theme={bootstrapSelectTheme}
               />
             </Form.Group>
             {this.props.type === 'edit' && (
               <Form.Group>
-                <Form.Label>Status</Form.Label>
-                <Form.Check
-                  id="status-input"
-                  name="status"
-                  type="switch"
-                  checked={this.state.isLocked}
-                  label={this.state.isLocked ? 'Locked' : 'Unlocked'}
-                  onChange={this.handleLockedStatusChange}
-                />
+                <Form.Label className="input-label">System Access</Form.Label>
+                <div style={{ display: 'flex', alignItems: 'center' }}>
+                  <Form.Check
+                    id="system-access-input"
+                    name="isLocked"
+                    type="switch"
+                    checked={this.state.isLocked}
+                    label={this.state.isLocked ? 'Locked' : 'Unlocked'}
+                    onChange={this.handleChange}
+                  />
+                  {this.state.lockReason === 'Auto-locked by the System' && this.state.isLocked && (
+                    <div className="locked-warning-text ml-2">
+                      <FontAwesomeIcon className="mr-1" icon={faExclamationCircle} />
+                      <span>{this.state.autoLockReason}</span>
+                    </div>
+                  )}
+                </div>
+              </Form.Group>
+            )}
+            {this.props.type === 'edit' && (
+              <Form.Group>
+                <Form.Label className="input-label" htmlFor="status-select">
+                  Status
+                </Form.Label>
+                <span data-for="disabled-status-select" data-tip="">
+                  <Select
+                    inputId="status-select"
+                    id="status"
+                    name="status"
+                    value={this.getStatusValue()}
+                    options={this.state.lockReasonOptions.map(lockReason => {
+                      return { label: lockReason, value: lockReason };
+                    })}
+                    onChange={data => this.setState({ lockReason: data.value })}
+                    styles={cursorPointerStyle}
+                    theme={bootstrapSelectTheme}
+                    isDisabled={!this.state.isLocked}
+                  />
+                </span>
+                {!this.state.isLocked && (
+                  <ReactTooltip id="disabled-status-select" multiline={true} type="dark" effect="solid" place="top" className="tooltip-container">
```

Comment:
tooltip looks great! I think I prefer having it placed on the bottom because there's more whitespace there

Reply:
Sure! Here is the update: [d2f14f7ed8eaadf3fd01b5e4a9174f88bc436f87](https://github.com/SaraAlert/SaraAlert/pull/1180/commits/d2f14f7ed8eaadf3fd01b5e4a9174f88bc436f87)